load_font

sakshamror opened the pull request:
# Description

This PR adds a working C++ SDL2 example that demonstrates how to load and display a TrueType font using SDL_ttf. The window renders the message “Font Loaded Successfully!” centered on a light gray background. This update provides a minimal, verifiable example for font rendering with error handling for font loading, surface creation, and texture generation.

Compile Command:g++ load_font-1-beyond-sk.cpp -o load_font-1-beyond-sk \
  -DSDL_MAIN_HANDLED \
  -IC:/msys64/mingw64/include/SDL2 \
  -LC:/msys64/mingw64/lib \
  -lSDL2 -lSDL2_ttf
  
## Type of change

- [x] New feature (non-breaking change which adds functionality)


## How Has This Been Tested?

-Compiled and run locally on Windows using SDL2 and SDL_ttf.
-Verified that the text is rendered correctly in the center of the window.
-Confirmed that the application closes cleanly when the user presses the close button on the window.

## Testing Checklist

- [ ] Tested in latest Chrome
- [ ] Tested in latest Firefox
- [ ] npm run build
- [ ] npm run preview

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code in hard-to-understand areas
- [x] My changes generate no new warnings


## Folders and Files Added/Modified

  - [ ] Added: usage-examples/sdl2/load_font-1-beyond.cpp

omckeon commented on public/usage-examples/graphics/load_font-1-example-beyond-sk.cpp:
```diff
@@ -0,0 +1,76 @@
+#define SDL_MAIN_HANDLED
```
Not needed

sakshamror replied:
Thank you for the feedback. I have removed the unnecessary define and updated the filename and PR title accordingly to reflect that this is for Beyond SplashKit.

omckeon commented on public/usage-examples/graphics/load_font-1-example-beyond-sk.cpp:
```diff
@@ -0,0 +1,76 @@
+    SDL_Renderer *renderer = SDL_CreateRenderer(window, -1, SDL_RENDERER_ACCELERATED);
+
+    // Load the font
+    TTF_Font *my_font = TTF_OpenFont("RobotoSlab-VariableFont_wght.ttf", 40);
```
This should be using the same filename/font as the splashkit version. It will also need to use the path `Resources/fonts/RobotoSlab.ttf`.

sakshamror replied:
Thanx for the feedback! I have updated the code accordingly.

omckeon commented on public/usage-examples/graphics/load_font-1-example-beyond-sk.cpp:
```diff
@@ -0,0 +1,76 @@
+    SDL_Color text_color = {0, 0, 0, 255};
+
+    // Render the text surface
+    SDL_Surface *text_surface = TTF_RenderText_Solid(my_font, "Font Loaded Successfully!", text_color);
```
This needs to match the splashkit version more closely. eg. "Hello, SDL2!"

sakshamror replied:
Thanx for the feedback! I have updated the text to match the SplashKit version.

omckeon commented on public/usage-examples/graphics/load_font-1-example-beyond-sk.cpp:
```diff
@@ -0,0 +1,76 @@
+    SDL_RenderCopy(renderer, text_texture, NULL, &text_rect);
+    SDL_RenderPresent(renderer);
+
+    SDL_Event event;
+    bool running = true;
+    while (running)
+    {
+        while (SDL_PollEvent(&event))
+        {
+            if (event.type == SDL_QUIT)
+            {
+                running = false;
+            }
+        }
+    }
```
Have a look at this page: https://splashkit.io/beyond-splashkit/graphics/0-getting-started-with-graphics/#open-window

Then fix this to be just a delay for 5 seconds (like the splashkit version).

sakshamror replied:
Thanx for the feedback! I have updated the code as suggested

omckeon commented on public/usage-examples/graphics/load_font-1-example-beyond-sk.cpp:
- Fix the file name (should be `load_font-1-example-beyond.cpp`, as mentioned in the CONTRIBUTING.mdx file.
- Update the comments to include more beyond splashkit comments, and make sure to have the same comments from the splashkit version as well.

sakshamror replied:
Thanx for the feedback! I have updated the code as suggested

omckeon reviewed (changes requested):
I've added some comments below for specific changes.

You also need to update the splashkit cpp file name to include "-sk" on the end.

And please improve your PR title to make it clear that this is for "Beyond SplashKit".

omckeon commented:
Also, @sakshamror, please update the description to include the command you used to compile this code.

sakshamror commented:
> Also, @sakshamror, please update the description to include the command you used to compile this code.

sakshamror commented:
> I've added some comments below for specific changes.
> 
> You also need to update the splashkit cpp file name to include "-sk" on the end.
> 
> And please improve your PR title to make it clear that this is for "Beyond SplashKit".

Thank you for the feedback, I have updated the code accordingly.
